lms/djangoapps/completion/tests/test_handlers.py

```diff
@@ -0,0 +1,88 @@
+"""
+Test signal handlers.
+"""
+
+from datetime import datetime
+
+import ddt
+from django.test import TestCase
+from mock import patch
+from opaque_keys.edx.keys import CourseKey
+from pytz import utc
+import six
+
+from lms.djangoapps.grades.signals.signals import PROBLEM_WEIGHTED_SCORE_CHANGED
+from student.tests.factories import UserFactory
+
+from .. import handlers
+from ..models import BlockCompletion
+
+
+@ddt.ddt
+class MockedHandlerTestCase(TestCase):
+    """
+    Test the signal handler
+    """
+
+    def setUp(self):
+        super(MockedHandlerTestCase, self).setUp()
+        self.user = UserFactory.create()
+        self.course_key = CourseKey.from_string("course-v1:a+valid+course")
+        self.block_key = self.course_key.make_usage_key(block_type="video", block_id="mah-video")
+
+    @ddt.data(
+        {'score_deleted': True, 'expected': 0.0},
+        {'score_deleted': False, 'expected': 1.0},
+        {'expected': 1.0},
+    )
+    def test_handler_submits_completion(self, params):
+        expected_completion = params.pop('expected')
+        handlers.scorable_block_completion(
+            sender=self,
+            user_id=self.user.id,
+            course_id=six.text_type(self.course_key),
+            usage_id=six.text_type(self.block_key),
+            weighted_earned=0.0,
+            weighted_possible=3.0,
+            modified=datetime.utcnow().replace(tzinfo=utc),
+            score_db_table='submissions',
+            **params
+        )
+        completion = BlockCompletion.objects.get(user=self.user, course_key=self.course_key, block_key=self.block_key)
+        self.assertEqual(completion.completion, expected_completion)
+
+    def test_handler_resets_completion_on_score_delete(self):
+        user = UserFactory.create()
+        course_key = CourseKey.from_string("course-v1:a+valid+course")
+        block_key = course_key.make_usage_key(block_type="video", block_id="mah-video")
+
+        handlers.scorable_block_completion(
+            sender=self,
+            user_id=user.id,
+            course_id=six.text_type(course_key),
+            usage_id=six.text_type(block_key),
+            weighted_earned=0.0,
+            weighted_possible=3.0,
+            modified=datetime.utcnow().replace(tzinfo=utc),
+            score_db_table='submissions',
+        )
+        completion = BlockCompletion.objects.get(user=user, course_key=course_key, block_key=block_key)
+        self.assertEqual(completion.completion, 1.0)
```

Comment:
This test doesn't seem to be doing what it says (`test_handler_resets_completion_on_score_delete`) and it seems redundant given the `ddt` variants of the `test_handler_submits_completion ` test above. Is this just a remnant from before a ddt refactor?

Reply:
Yeah.  The DDT was a late addition, and I think I did it half way through copy-pasting this test (before revising it to match the name.